x/model/keeper/msg_server_model_version.go

```diff
@@ -109,15 +104,35 @@ func (k msgServer) UpdateModelVersion(goCtx context.Context, msg *types.MsgUpdat
 		return nil, types.NewErrMaxSVLessThanMinSV(modelVersion.MinApplicableSoftwareVersion, msg.MaxApplicableSoftwareVersion)
 	}
 
-	// update existing model version value only if corresponding value in MsgUpdate is not empty
+	if msg.OtaUrl != "" && modelVersion.OtaUrl != "" && msg.OtaUrl != modelVersion.OtaUrl {
```

Comment:
The check seems wrong. Why do we need it? OtaURL should be always mutable without any restrictions.